website/docs/language/checks/index.mdx

```diff
@@ -0,0 +1,114 @@
+
+The [previous example](#checks-syntax) would fail during any initial planning, as the configuration hasn't been applied so the website simply would not exist. As written, this would be expected behaviour but is not particularly useful and is potentially noisy as other genuine errors in the configuration could be obscured by the warning message.
+
+An addition of the `depends_on` attribute, linking the scoped data source to the highest-levelled piece of infrastructure such as a load balancer, would mean that during any initial planning the check would instead report as `known after apply`. In addition, it would not produce any unnecessary or noisy warnings. Going forward, the check would execute again for both plans and applies providing the required validation.
+
+One caveat to the `depends_on` attribute is that any changes to the dependencies would again cause the check block to report as `known after apply`, essentially meaning the check block would not execute during that planning stage. Depending on your use case this behavior could be acceptable or problematic.
+
+### Assertions
+
+Check blocks validate your custom assertions using `assert` blocks. Each `check` block must have at least one, but potentially many, `assert` blocks. Each `assert` block has a [`condition` attribute](/terraform/language/expressions/custom-conditions#condition-expressions) and an [`error_message` attribute](/terraform/language/expressions/custom-conditions#error-messages).
+
+Unlike other [custom conditions](/terraform/language/expressions/custom-conditions), assertions do not affect Terraform's execution of an operation. A failed assertion reports a warning without halting the ongoing operation. This contrasts with other custom conditions, such as a postcondition, where Terraform produces an error immediately, halting the operation and blocking the application or planning of future resources.
+
+Condition arguments within `assert` blocks can refer to scoped data sources within the enclosing `check` block and any variables, resources, data sources, or module outputs within the current module.
+
+[Learn more about assertions](/terraform/language/expressions/custom-conditions#check-assertions).
+
+### Meta-Arguments
+
+Check blocks do not currently support [meta-arguments](/terraform/language/resources/syntax#meta-arguments). We are still collecting feedback on this feature, so if your use case would benefit from check blocks supporting meta-arguments, please [let us know](https://github.com/hashicorp/terraform/issues/new/choose).
+
+## Continuous Validation in Terraform Cloud
+
+Terraform Cloud can automatically validate whether checks in a workspace’s configuration continue to pass after Terraform provisions new infrastructure. See [Continuous Validation](/terraform/cloud-docs/workspaces/health#continuous-validation) for details.
+
+## Choosing Checks or other Custom Conditions
```

Comment:
@rkoron007, I've added a lot to this section. I think it's worth a fresh review. Just from here to the end of the file.